frontend/app/dashboard/page.jsx

```diff
@@ -1,38 +1,56 @@
 'use client';
-import { useEffect, useState } from 'react';
+import { useEffect } from 'react';
 import { onAuthStateChanged } from 'firebase/auth';
 import { useRouter } from 'next/navigation';
 import { auth } from '../../firebase/config';
 import Cookies from 'js-cookie';
+import { client, FIND_USER, CREATE_USER } from '../../gql';
 
 const Dashboard = () => {
     const router = useRouter();
-    const [ userData, setUserData ] = useState();
-
+    
     useEffect(() => {
-            onAuthStateChanged(auth, user => {
-                if (user) {
-                    const uidVerify = Cookies.get('uid');
-                    if (uidVerify !== user.uid) {
-                        Cookies.remove('uid');
-                        router.push('/')
-                    } else {
-                        setUserData(user);
+        const unsubscribe = onAuthStateChanged(auth, user => {
+            const cookieUid = Cookies.get('uid');
+            // If the user doesn't exist or the users uid stored doesn't match
+            // firebases uid then remove the cookie and redirect the user to login page
+            if(!user || cookieUid !== user.uid) {
+                Cookies.remove('uid');
+                router.push('/');
+            }
+
+            if (user) {
+                const { uid } = user;
+                // Check if user exists in database
```

Comment:
I LOVE comments. These comments however are made to explain complex code, and this block from lines 13 - 49 is now at a point that it can be refactored. If you simplify the code by chunking it and using descriptive variables, you're future proofing this block and you won't need the comments. For example:

```
if (user) {
        const { uid, email } = user;

        const user = await getUser(uid);
        const hasAirtableProfile = user.data.users.length !== 0;

        if (!hasAirtableProfile) {
          await createUser(email);
        }
      }
```
And then either at the top of the file or in a separate airtable utils file (wtvr you prefer):
```
const createUser = email => {
  client.mutate({
    mutation: CREATE_USER,
    variables: { email, uid },
    // Update apollo cache with new user created
    refetchQueries: [
      {
        query: FIND_USER,
        variables: { uid },
      },
    ],
  });
};

const getUser = uid =>
  client.query({
    query: FIND_USER,
    variables: { uid },
  });
```